### PR TITLE
Reworks sentient swords

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
Reworks sentient swords, gives them some new abilities 
Sentient swords can now attack in melee range like someone can with any weapon
Sentient swords now deal 12 normal damage, and 7 while attacking by themselves while not in someone's hand, deals normal force while in someone's hand. Both people can attack separately
Like most other objects, the sentient sword can now be harmed by melee items as well
Sentient swords now only have 100 integrity 
Sentient swords can be fully "healed" by using a soulshard on them
Sentient swords now have a "lunge" ability. This throws the sentient sword at a clicked target in their viewrange. Cooldown of 15 seconds.
TODO:

- [x] Unjank the code
- [x] Give it a "lunge" ability with a cooldown
- [x] Balance it
- [x] Give the ability for melee items to damage this and blobs
~~- [ ] Maybe add a sentient sword based midround (might do this in a different pr but it's been living rent free in my head)~~ Ima do this in another PR
- [x] polish it 
- [x] add an actual video of it in action
## Why It's Good For The Game
Right now sentient sword is bad to play and bad to use. You're a discount pAI in sword form and imminently gets yeeted into the nearest chasm once a miner realizes this. Making this actually fun to play is the priority and this PR should do that by giving you helpful options in combat and a niche movement option. 

## Video of changes

https://user-images.githubusercontent.com/96800819/224456583-2a592b8f-8da3-4cf9-9797-58fe40e329de.mp4

## Testing
it works currently
## Changelog
:cl: GDN, ATP engineer
tweak: Sentient sword has been reworked
/:cl:

### CODE REVIEW

File: config/example/config.toml
Comment (GDNgit) [inline config/example/config.toml:324]: Don't worry about the codeowner review AA, this is enabled just for local testing

**Comment by Peakwastaken** [comment]
It can yeet itself ? Hilarious, love it.

I'd rather switch the two numbers around though, fifteen for the person's slash and ten for the sword's slash, and it shoould maybe not be able to attack while not in the users hand as grump said on discord, besides that looks good !

**Comment by GDNgit** [comment]
> It can yeet itself ? Hilarious, love it.
> 
> I'd rather switch the two numbers around though, fifteen for the person's slash and ten for the sword's slash, and it shoould maybe not be able to attack while not in the users hand as grump said on discord, besides that looks good !

Don't think I mentioned it actually in the pr but
1. Sword only deals 10 damage on the ground, and can be attacked like a mob (unlike other items)
2. I didn't want to make it deal 15 in hand because it would make the weapon automatically outclass other nullrod options, but I can change it to 12 damage by the person and 10 while in hand (5 on the ground) when I get up tomorrow
Thanks for the feedback btw!

**Comment by GDNgit** [comment]
Oh and to anyone worried about potential blob cheese, it won't work. Sword instantly will get destroyed (boo womp)

**Comment by GDNgit** [comment]
> It can yeet itself ? Hilarious, love it.
> 
> I'd rather switch the two numbers around though, fifteen for the person's slash and ten for the sword's slash, and it shoould maybe not be able to attack while not in the users hand as grump said on discord, besides that looks good !

Damage has been brought down to 12 from 15

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (S34NW) [inline code/game/objects/items/weapons/holy_weapons.dm:309]:
```suggestion
	if(!istype(I, /obj/item/soulstone) || !possessed)
		return ..()
```

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (S34NW) [inline code/game/objects/items/weapons/holy_weapons.dm:320]:
```suggestion
```

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (S34NW) [inline code/game/objects/items/weapons/holy_weapons.dm:319]:
```suggestion
	if(!(obj_integrity < max_integrity))
		to_chat(user, "<span class='notice'>You have no reason to replace a perfectly good soulstone with a new one.</span>")
		return
	to_chat(user, "<span class='notice'>You load a new soulstone into the possessed blade.</span>")
	playsound(user, 'sound/weapons/gun_interactions/shotgunpump.ogg', 60, 1)
	obj_integrity = max_integrity
	for(var/mob/living/simple_animal/shade/sword/sword_shade in contents)
		sword_shade.health = sword_shade.maxHealth
	qdel(I)
```

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (S34NW) [inline code/game/objects/items/weapons/holy_weapons.dm:343]:
```suggestion
	if(ismob(loc))
		var/mob/living/carbon/human/our_location = loc
		if(istype(our_location))
			if(!(src == our_location.l_hand) && !(src == our_location.r_hand))
				return
			if(our_location.Adjacent(attacking_atom)) // with a buddy we deal 10 damage :D
				our_location.do_attack_animation(attacking_atom, used_item = src)
				melee_attack_chain(attacking_shade, attacking_atom)
			return
```
No need for a is not turf AND is mob check. Just the mob one is fine.

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (S34NW) [inline code/game/objects/items/weapons/holy_weapons.dm:360]:
```suggestion
	if(client?.click_intercept)
```

**Review by S34NW — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (GDNgit) [reply code/game/objects/items/weapons/holy_weapons.dm:360]: if they don't have a client but clicked on something we want to know that's the case

Comment (S34NW) [reply code/game/objects/items/weapons/holy_weapons.dm:382]: Fair

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (lewcc) [inline code/game/objects/items/weapons/holy_weapons.dm:272]: Going to guess you probably want to remove this

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (lewcc) [inline code/game/objects/items/weapons/holy_weapons.dm:336]: Tactical reload, I love it

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (lewcc) [inline code/game/objects/items/weapons/holy_weapons.dm:351]:
```suggestion
	if(!ismovable(attacking_atom))
```

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (lewcc) [inline code/game/objects/items/weapons/holy_weapons.dm:357]:
```suggestion
			if(src != our_location.l_hand && src != our_location.r_hand)
```

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (lewcc) [inline code/game/objects/items/weapons/holy_weapons.dm:332]:
```suggestion
	if(obj_integrity >= max_integrity)
```

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (lewcc) [inline code/game/objects/items/weapons/holy_weapons.dm:366]: Why a prob(40) here? What's this accomplishing?

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (GDNgit) [reply code/game/objects/items/weapons/holy_weapons.dm:272]: I *kinda* wanna leave it in for further testing just because of how much of a bitch it is rn, although it can be removed if that's a demand

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (GDNgit) [reply code/game/objects/items/weapons/holy_weapons.dm:366]: I figure this just makes more sense, animals that aren't sentient need a way to attack to prevent cheese and they're probably not gonna pick up on the fact that the sword is sentient, just that it is hurtin em, so there's a decently high chance they attack it

Comment (lewcc) [reply code/game/objects/items/weapons/holy_weapons.dm:366]: Mind adding a comment to explain that?

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (lewcc) [reply code/game/objects/items/weapons/holy_weapons.dm:272]: Fair enough, I don't think it *needs* to go, especially if it keeps things easier to work with. Just might be a demand before merge.

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (GDNgit) [reply code/game/objects/items/weapons/holy_weapons.dm]: Changes pushed

**Review by S34NW — APPROVED** [review]
my concerns are addressed, I can't see anything glaringly wrong now

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (Henri215) [inline code/game/objects/items/weapons/holy_weapons.dm:315]:
```suggestion
	playsound(user, 'sound/weapons/gun_interactions/shotgunpump.ogg', 60, TRUE)
```

File: code/datums/spells/sentient_sword_lunge.dm
Comment (Henri215) [inline code/datums/spells/sentient_sword_lunge.dm:3]:
```suggestion
	desc = "Lunge at something in your view range."
```

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (hal9000PR) [inline code/game/objects/items/weapons/holy_weapons.dm:342]: early return if the `get_turf(src)` isn't adjancent to `attacking_atom` rather than checking it twice, here and on line 338.

**Review by hal9000PR — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (GDNgit) [reply code/game/objects/items/weapons/holy_weapons.dm:342]: won't that just make it so that if you are in an object with the sword you can attack through the object? Like a mecha for example

Comment (SteelSlayer) [reply code/game/objects/items/weapons/holy_weapons.dm:342]: Shouldn't be any logic change, just an early return.

Comment (SteelSlayer) [reply code/game/objects/items/weapons/holy_weapons.dm:342]: Discussed on discord a bit. The concern is that calling `Adjacent` on a turf would make you able to attack through objects, However `atom/movable/Adjacent` (which is called from this parent) also calls `Adjacent` on a turf, so I'm not there will be any weird behavior here.

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (Henri215) [inline code/game/objects/items/weapons/holy_weapons.dm:349]:
```suggestion
		force -= 5
		var/mob/living/simple_animal/hostile/hostile_target = attacking_atom
		if(istype(hostile_target) && prob(40)) // Cheese reduction, non sentient animals have a hard time attacking things in objects
			attack_animal(hostile_target)
		do_attack_animation(attacking_atom, used_item = src)
		melee_attack_chain(attacking_shade, attacking_atom)
		force += 5
```

File: code/datums/spells/sentient_sword_lunge.dm
Comment (GDNgit) [reply code/datums/spells/sentient_sword_lunge.dm:16]: You can't cast while outside your blade, not your users hand

Comment (Henri215) [reply code/datums/spells/sentient_sword_lunge.dm:16]: By blade you mean..its blade mode?

Comment (GDNgit) [reply code/datums/spells/sentient_sword_lunge.dm:16]: It's a mob in an object, if it gets removed it shouldn't be able to lunge

**Comment by Venuska1117** [comment]
Need more ideas?
here if it help (most of it are concepts)
https://www.paradisestation.org/forum/topic/23972-vens-idea-how-to-make-sentient-sword-be-more-worthly/

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (GDNgit) [reply code/game/objects/items/weapons/holy_weapons.dm]: This breaks shit

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (Sirryan2002) [reply code/game/objects/items/weapons/holy_weapons.dm:272]: this should be removed before merge unless it's converted into an actual usable tool (which probably won't be used tbqh). Stuff shouldn't be left in that only codermins can access with magic.

File: code/game/objects/items/weapons/holy_weapons.dm
Comment (Sirryan2002) [inline code/game/objects/items/weapons/holy_weapons.dm:355]:
```suggestion
	if(ishuman(loc))
```
should typecheck to what var typepath you're using

Comment (GDNgit) [reply code/game/objects/items/weapons/holy_weapons.dm]: changed

**Review by Sirryan2002 — APPROVED** [review]

**Review by SteelSlayer — APPROVED** [review]

**Comment by Sirryan2002** [comment]
Awaiting Merge, hal's CR needs to be dismissed

**Pull request merged**
